src/parser/mod.rs

```diff
@@ -1676,7 +1676,13 @@ impl<'a, 'o, 'c> Parser<'a, 'o, 'c> {
         }
 
         *text = text[end..].to_vec();
-        let checkbox = inlines::make_inline(self.arena, NodeValue::TaskItem(active));
+        let checkbox = inlines::make_inline(
+            self.arena,
+            NodeValue::TaskItem {
+                checked: symbol != b" ",
+                symbol: symbol[0] as char,
```

Comment:
Because we're only matching a single byte in the regex, I think we are better off leaving this field as `u8` rather than casting it to `char` (which could imply a full set of Unicode characters might be stored here). This keeps us consistent with the other stored characters, `bullet_char` and `fence_char`.

Reply:
Considering if we should rename it to `task_char`?

Reply:
We certainly could! I don't mind either way.